src/compiler.jl

```diff
@@ -6072,10 +6072,8 @@ function to_tape_type(Type::LLVM.API.LLVMTypeRef)::Tuple{DataType,Bool}
             return UInt32, false
         elseif N == 64
             return UInt64, false
-        elseif N == 128
-            return UInt128, false
         else
-            error("Can't construct tape type for integer of width $N")
+            return NTuple{Int(N), UInt8}, false
```

Comment:
N/8 presumably, but per other comment I'm somewhat worried about this

Reply:
I think you would have seen that before since Julia 1.9.2 and earlier will have LLVM assert on the i128 like in my reply to @vchuravy. The other approach is to keep the UInt128 branch and accept failing on 1.9.2 and below.

Reply:
let's try to leave the i128 version and track down and fix the assertion

Reply:
It's this https://github.com/llvm/llvm-project/issues/23879

Reply:
Hm okay in whch case yeah I think I'd prefer to version gate the test personally.

Reply:
Also can we restore the old error if other int type size, also N should be N == 128 not N = 128 here

Reply:
The inline string looks i512 or i1024 and we break here even on Julia 1.10 and nightly. These types can be defined by e.g., primitive type VeryLongInt <: Integer 512 end
so I don't think an error is the right thing here.

Reply:
I mean the thing that matters here is that convert(LLVMType, output) must be the same llvm type that this takes as an input. So if we return VeryLongInt for an i512, VeryLongInt must lower to an i512

Reply:
Are the CI failures for type preservation related to this PR?